Content.Server/GameObjects/Components/Atmos/GasAnalyzerComponent.cs

```diff
@@ -49,16 +51,36 @@ public override ComponentState GetComponentState()
 
         /// <summary>
         /// Call this from other components to open the gas analyzer UI.
+        /// Uses the player position.
         /// </summary>
+        /// <param name="session">The session to open the ui for</param>
         public void OpenInterface(IPlayerSession session)
         {
+            _checkPlayer = true;
+            _position = null;
+            _userInterface.Open(session);
+            UpdateUserInterface();
+            Resync();
+        }
+
+        /// <summary>
+        /// Call this from other components to open the gas analyzer UI.
+        /// Uses a given position.
+        /// </summary>
+        /// <param name="session">The session to open the ui for</param>
+        /// <param name="pos">The position to analyze the gas</param>
+        public void OpenInterface(IPlayerSession session, GridCoordinates pos)
```

Comment:
Not quite what I meant.

I meant only have one `OpenInterface` method, that takes the sesh and the position.

For the player's pos, `OpenInterface(actor.playerSession, Owner.Transform.GridPosition);`
For a clicked pos, `OpenInterface(actor.playerSession, eventArgs.ClickLocation);

Then also get rid of `_checkPlayer` since it won't matter, you'll just have 1 pos field that's either a clicked tile or the user's tile and it doesn't need any special handling for either.

Reply:
Well I wanted the playerpos to always update

Reply:
Sure, that's fine then.